File: rust/src/models/package_pricing.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct PackagePricing {
    pub block_size: i32,

    pub rate: rust_decimal::Decimal,
}

impl PackagePricing {
    pub fn new(block_size: i32, rate: rust_decimal::Decimal) -> Self {
        Self { block_size, rate }
    }
}

File: typescript/src/models/packagePricing.ts
// this file is @generated

export interface PackagePricing {
  blockSize: number;

  rate: string;
}

export const PackagePricingSerializer = {
  _fromJsonObject(object: any): PackagePricing {
    return {
      blockSize: object["block_size"],
      rate: object["rate"],
    };
  },

  _toJsonObject(self: PackagePricing): any {
    return {
      block_size: self.blockSize,
      rate: self.rate,
    };
  },
};
